### PR TITLE
Use VA_OPT to visit empty structs as well

### PR DESCRIPTION
**Pull request by 1-Harshit**
As discussed earlier @cbeck88, raising a patch here as well.

TL;DR
The current setup doesn't work with empty struct. Patched solution:
GCC godbolt: https://godbolt.org/z/44T1ndTbh
MSVC godbolt: https://godbolt.org/z/ozxfYa7PW

Full explanation:
There's a catch to VISIT_STRUCT_PP_MAP. Ideally, it should do nothing if no arguments are provided, but the build crashes for such cases. The underlying issue is:
- VISIT_STRUCT_PP_MAP(f) expands to VISIT_STRUCT_APPLY_F_(VISIT_STRUCT_CONCAT(VISIT_STRUCT_APPLYF, VISIT_STRUCT_PP_NARG()), f,)
- Which in turn should become VISIT_STRUCT_APPLYF0(f) [Note that this 0 is the resolution of macro VISIT_STRUCT_PP_NARG()]
- But it expands to VISIT_STRUCT_APPLYF1(f,)

This, in turn, causes a compilation issue for my use case. I investigated the issue, and here is the problem:
-  For VISIT_STRUCT_PP_NARG(ONE), the macro expands as VISIT_STRUCT_PP_ARG_N(ONE, 69, 68, ...) -> Where the Nth term becomes 1, and we get 1 [Correct behaviour]
- For VISIT_STRUCT_PP_NARG() [no args] the macro expands as VISIT_STRUCT_PP_ARG_N(, 69, 68, ...) -> Where also the Nth term becomes 1, and we get 1 [Incorrect behaviour]

This happens as the empty field before the comma is considered an argument to the macro. ++23 introduces a solution for such cases with the reserved macro [__VA_OPT__](https://en.cppreference.com/w/cpp/preprocessor/replace#:~:text=For%20versions%20(3%2C4)%2C%20replacement%2Dlist%20may%20contain%20the%20token%20sequence%20__VA_OPT__).

Now VISIT_STRUCT_PP_NARG() resolve to 0 in cases where no arguments are provided. I’ve included two Godbolt links comparing the original and patched versions for your reference [original Godbolt](https://godbolt.org/z/P1sdT88P3) vs. [patched Godbolt](https://godbolt.org/z/4v8WohqKf).

### CODE REVIEW

**Review by cbeck88 — APPROVED** [review]
it looks good to me, thank you!

**Comment by 1-Harshit** [comment]
@cbeck88 I don't have write access to the repository to merge these changes
image

**Comment by 1-Harshit** [comment]
@cbeck88 should we update the VISIT_STRUCT_VERSION_PATCH to 1 ?

**Comment by cbeck88** [comment]
> @cbeck88 should we update the VISIT_STRUCT_VERSION_PATCH to 1 ?

Sure that seems good

**Comment by cbeck88** [comment]
> @cbeck88 I don't have write access to the repository to merge these changes image

Sorry, I thought it would let you merge with approval. I’ll merge shortly

**Comment by 1-Harshit** [comment]
@cbeck88 update patch version

**Comment by 1-Harshit** [comment]
@cbeck88 can you please create a github release for this?
